gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+/*!
+GPII download and save solutions registry from source code repositoyr
+
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    kettle = require("kettle"),
+    jqUnit = fluid.registerNamespace("jqUnit"),
+    gpii = fluid.registerNamespace("gpii");
+
+kettle.loadTestingSupport();
+
+fluid.require("%flowManager/src/SolutionsRegistryDataSource.js");
+fluid.registerNamespace("gpii.tests.solutionsRegistry");
+
+gpii.tests.solutionsRegistry = {
+    path: "%gpii-universal/testData/solutions",
+    platformIds: ["win32", "linux", "darwin", "android", "web"],
+    platformId: "darwin",
+    repositoryDarwinSolutions: require("./data/darwin.json5"),
+    requestOptions: { os: "darwin" },
+    win32PlatformId: "win32"
+};
+
+// Check that the given platform id has been stripped out of the result.
+gpii.tests.solutionsRegistry.checkOneRegistry = function (msg, platformId, result) {
+    jqUnit.assertValue(msg, result);
+    jqUnit.assertFalse(msg, platformId in result);
+};
+
+// equalityCheck is either jqUnit.assertDeepEq() or jqUnit.assertDeepNeq().
+gpii.tests.solutionsRegistry.checkNamedRegistry = function (msg, equalityCheck, expectedSolutions, result) {
+    jqUnit.assertValue(msg, result);
+    equalityCheck(msg, expectedSolutions, result);
+};
+
+// ======== Testing the solutions registry data source used by the CBFM ========
+
+fluid.defaults("gpii.tests.solutionsRegistry.cloud.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    components: {
+        solutionsRegistryDataSource: {
+            type: "gpii.flowManager.solutionsRegistry.dataSource.cloud",
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
+        name: "Solutions registry used by CBFM - file-based solutions",
+        expect: 2,
+        tests: [{
+            name: "Solutions registry used with CBFM",
+            sequence: [
+                { funcName: "fluid.log", args: ["Solutions Registry for CBFM, getting named registry -- START"] },
+
+                // For some reason, even though the solutionsRegistryReady event
```

Comment:
Shall these comments and commented code be removed?

Reply:
I was hoping the reviewers would have some insights here :-)

I currently think that by the time the tests are marshalled and ready to go, the event is long  gone, but I'm not sure.  In any case, I'll remove the comments and code now.

Reply:
Also, since I now understand the `task/resolve/reject` pattern in the test fixtures better, I've rewritten them to no longer depend on the `onRead` event.  Using the `task` pattern to check the promise returned by a `SRDS.get(...)` is what is needed here.